backend/de.metas.purchasecandidate.base/src/main/java/de/metas/purchasecandidate/material/event/PurchaseCandidateAdvisedEventCreator.java

```diff
@@ -70,18 +81,33 @@ public Optional<PurchaseCandidateAdvisedEvent> createPurchaseAdvisedEvent(
 			return Optional.empty();
 		}
 
-		final I_PP_Product_Planning productPlanning = mrpContext.getProductPlanning();
-
-		final PurchaseCandidateAdvisedEvent event = PurchaseCandidateAdvisedEvent
-				.builder()
+		final PurchaseCandidateAdvisedEvent.PurchaseCandidateAdvisedEventBuilder event = PurchaseCandidateAdvisedEvent.builder()
 				.eventDescriptor(EventDescriptor.ofEventDescriptor(supplyRequiredDescriptor.getEventDescriptor()))
-				.supplyRequiredDescriptor(supplyRequiredDescriptor)
 				.directlyCreatePurchaseCandidate(productPlanning.isCreatePlan())
 				.productPlanningId(productPlanning.getPP_Product_Planning_ID())
-				.vendorId(defaultVendorProductInfo.get().getVendorId().getRepoId())
-				.build();
+				.vendorId(defaultVendorProductInfo.get().getVendorId().getRepoId());
+
+		if(productPlanning.isLotForLot())
+		{
+			final BigDecimal fullDemandQty = supplyRequiredDescriptor.getFullDemandQty();
+			final MaterialDescriptor updatedMaterialDescriptor = supplyRequiredDescriptor.getMaterialDescriptor().withQuantity(fullDemandQty);
+			event.supplyRequiredDescriptor(supplyRequiredDescriptor.toBuilder()
+												   .materialDescriptor(updatedMaterialDescriptor)
+												   .isLotForLot("Y")
```

Comment:
why not using true/false?

Reply:
because the displayed Info can be none, yes, no